### PR TITLE
docs(components): Adds and expands on deprecation.

### LINKED ISSUE #3983: Improve documentation around deprecation
Need info about deprecation - how long a component will be deprecated, etc
Maybe can put this info in the deprecated alert.

### PR DESCRIPTION
**Pull request by edonehoo**
Closes #3983

### CODE REVIEW

File: packages/documentation-framework/templates/mdx.js [hunk 1]
```diff
@@ -69,13 +69,13 @@ const MDXChildTemplate = ({
...
         <InlineAlert title="Deprecated feature" variant="warning">
-          This implementation has been deprecated in favor of a newer implementation, and is no longer getting maintained or enhanced.
+          This component implementation has been deprecated in favor of a newer solution, and is no longer being maintained or enhanced.
           {newImplementationLink && (
             <React.Fragment>
               You can find the <Link to={newImplementationLink}>updated implementation here</Link>.
             </React.Fragment>
           )}
-          {' '}To learn more about the process, visit our <Link to="/get-started/about-patternfly#patternfly-release-cadence">about page</Link>.
+          {' '}To learn more about the deprecated components, visit <Link to="/get-started/about-patternfly#deprecated-components">about PatternFly.</Link>
```
Comment (edonehoo) [inline packages/documentation-framework/templates/mdx.js:78]:
```suggestion
          {' '}To learn more about deprecated components, visit <Link to="/get-started/about-patternfly#deprecated-components">about PatternFly.</Link>
```

**Comment by edonehoo** [comment]
Not sure how in depth we'd like to go on org? Made some small additions and edits to some of the existing messaging. Also added a note for chip. 

Eventually, this content could find a home in a different area based on [#1301](https://github.com/patternfly/patternfly-design/issues/1301)

**Review by andrew-ronaldson — APPROVED** [review]
Awesome!

File: packages/documentation-framework/templates/mdx.js [hunk 2]
```diff
@@ -69,13 +69,13 @@ const MDXChildTemplate = ({
...
         <InlineAlert title="Deprecated feature" variant="warning">
-          This implementation has been deprecated in favor of a newer implementation, and is no longer getting maintained or enhanced.
+          This component implementation has been deprecated in favor of a newer solution, and is no longer being maintained or enhanced.
           {newImplementationLink && (
             <React.Fragment>
               You can find the <Link to={newImplementationLink}>updated implementation here</Link>.
             </React.Fragment>
           )}
-          {' '}To learn more about the process, visit our <Link to="/get-started/about-patternfly#patternfly-release-cadence">about page</Link>.
+          {' '}To learn more about deprecated components, visit <Link to="/get-started/about-patternfly#deprecated-components">about PatternFly.</Link>
```
Comment (srambach) [inline packages/documentation-framework/templates/mdx.js:78]: This anchor doesn't seem to be working - it links to the top of the page.

**Review by nicolethoen — APPROVED** [review]
I'm not sure why the anchor doesn't seem to be working - that should be a browser feature... looks good to me otherwise!

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**